app/common/CustomThemeProvider.tsx

```diff
@@ -0,0 +1,53 @@
+"use client";
+import {
+  SDSLightAppTheme,
+  SDSDarkAppTheme,
+  makeThemeOptions,
+} from "@czi-sds/components";
+import {
+  createTheme,
+  ThemeProvider as MuiThemeProvider,
+  Theme,
+} from "@mui/material";
+import { ReactNode, useState, useEffect } from "react";
+import { customTheme } from "./theme";
+
+export type ThemeMode = "light" | "dark";
+
+// CustomThemeProvider component to wrap your app
+export const CustomThemeProvider = ({ children }: { children: ReactNode }) => {
+  const [theme, setTheme] = useState<Theme>();
+
+  const updateTheme = (themeMode: ThemeMode) => {
+    const baseTheme =
+      themeMode === "light" ? SDSLightAppTheme : SDSDarkAppTheme;
+    const themeOptions = { ...baseTheme, ...customTheme };
+
+    const appTheme = makeThemeOptions(themeOptions, themeMode);
+    setTheme(createTheme(appTheme));
+  };
+
+  useEffect(() => {
+    const mediaQuery = window.matchMedia("(prefers-color-scheme: dark)");
+
+    // Set initial theme based on system preference
+    updateTheme(mediaQuery.matches ? "dark" : "light");
+
+    const handleThemeChange = (e: MediaQueryListEvent) => {
+      updateTheme(e.matches ? "dark" : "light");
+    };
+
+    // Listen for theme changes
+    mediaQuery.addEventListener("change", handleThemeChange);
+
+    // Clean up event listener on unmount
+    return () => {
+      mediaQuery.removeEventListener("change", handleThemeChange);
+    };
+  }, []);
+
+  if (!theme) {
+    return null;
```

Comment:
should we return `<>{children}</>` just in case 😁 ? Otherwise it seems like nothing will be rendered!

Reply:
yep, it will be ugly, but better to render it